com.onesignal.unity.core/Runtime/Notifications/Models/INotification.cs

```diff
@@ -215,57 +215,57 @@ public enum LockScreenVisibility {
         /// https://developer.apple.com/documentation/usernotifications/unnotificationserviceextension
         /// </summary>
         /// <remarks>iOS only</remarks>
-        public bool mutableContent;
+        bool MutableContent { get; }
 
         /// <summary>
         /// iOS Notification category key previously registered to display with
         /// </summary>
         /// <remarks>iOS only</remarks>
-        public string category;
+        string Category { get; }
 
         /// <summary>
         /// The badge number assigned to the application icon
         /// </summary>
         /// <remarks>iOS only</remarks>
-        public int badge;
+        int Badge { get; }
 
         /// <summary>
         /// The amount to increment the badge icon number
         /// </summary>
         /// <remarks>iOS only</remarks>
-        public int badgeIncrement;
+        int BadgeIncrement { get; }
 
         /// <summary>
         /// Groups notifications into threads
         /// </summary>
         /// <remarks>iOS 10+ only</remarks>
-        public string threadId;
+        string ThreadId { get; }
 
         /// <summary>
         /// Relevance Score for notification summary
         /// </summary>
         /// <remarks>iOS 15+ only</remarks>
-        public double relevanceScore;
+        double RelevanceScore { get; }
 
         /// <summary>
         /// Interruption Level of the notification
         /// </summary>
         /// <remarks>iOS 15+ only</remarks>
-        public string interruptionLevel;
+        string InterruptionLevel { get; }
 
         /// <summary>
         /// Attachments sent as part of the rich notification
         /// </summary>
         /// <remarks>iOS 10+ only</remarks>
-        public Dictionary<string, object> attachments;
+        Dictionary<string, object> Attachments { get; }

```

Comment:
you might be able to make this IDictionary<string, object> to provide a little more abstraction